x-pack/plugin/gpu/src/main/java/org/elasticsearch/xpack/gpu/GPUPlugin.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.gpu;
+
+import org.apache.lucene.codecs.KnnVectorsFormat;
+import org.apache.lucene.util.hnsw.HnswGraphBuilder;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.util.FeatureFlag;
+import org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapper;
+import org.elasticsearch.index.mapper.vectors.VectorsFormatProvider;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.plugins.internal.InternalVectorFormatProviderPlugin;
+import org.elasticsearch.xpack.gpu.codec.ES92GpuHnswSQVectorsFormat;
+import org.elasticsearch.xpack.gpu.codec.ES92GpuHnswVectorsFormat;
+
+import java.util.List;
+
+public class GPUPlugin extends Plugin implements InternalVectorFormatProviderPlugin {
+
+    public static final FeatureFlag GPU_FORMAT = new FeatureFlag("gpu_format");
```

Comment:
Should we give this feature flag a better name, say `gpu_indexing` or something?